public/javascripts/controllers.js

```diff
@@ -1,692 +1,816 @@
-      extra_fields: {'SUM': 'Count'}
-    },
-    'area': {
-      type: 'area',
-      field1: 'X-Axis',
-      field2: 'Y-Axis',
-      extra_fields: {'SUM': 'Count'}
-    },
-    'graph': {
-      type: 'graph',
-      field1: 'From',
-      field2: 'To',
-      multiple: true
-    },
-    'narratives': {
-      type: 'narratives',
-      field1: 'Date',
-      field2: 'People',
-      multiple: true
-    }
-  };
 
-  function setDefaults(){
-    $scope.visualizations.forEach(function(item){
-      if(item.maxresults == undefined || item.maxresults == null || item.maxresults == ""){
-        item.maxresults = "250";
+      $scope.save = function(){
+        setDefaults();
+
+        $http.post('/api/visualizations/save', {type: type, visualizations: $scope.visualizations})
+            .success(function(data){
+              if(!data.error){
+                $modalInstance.close('saved');
+              }
+            });
+      }
+    })
+
+    .controller('userOverviewCtrl', function($scope, $modal, $http){
+
+      $scope.itemsByPage = 10;
+      $scope.showNumberOfPages = 7;
+      $scope.user = {
+        isDeleted: 0
+      };
+      $scope.isMessage = false;
+      $scope.message = [];
+      $scope.isError = false;
+      $scope.error = [];
+
+      function loadUsers() {
+        $scope.rowCollection = [];
+        $http.post('/api/get_users')
+            .success(function (data) {
+
+              // Check if error
+              if (typeof data.error !== 'undefined') {
+                $scope.error.push(data.error.message);
+                $scope.isError = true;
+              }else if (data.response.hits.total > 0) {
+                //Load users in table
+                angular.forEach(data.response.hits.hits, function (value, key) {
+                  var u = value._source;
+                  if(u.isDeleted == 0) {
+                    $scope.rowCollection.push({
+                      id: value._id, firstName: u.firstName, lastName: u.lastName, email: u.email,
+                      role: 'N/A'
+                    });
+                  }
+                });
+                return data.response.hits.hits;
+              }
+
+              return false;
+            });
       }
-    });
-  };
 
+      var users = loadUsers();
 
-  $scope.save = function(){
-    setDefaults();
+      // Modal for adding new users
+      $scope.createUserModal = function(type){
+        $scope.modalInstance = $modal.open({
+          templateUrl: 'views/modals/createUser',
+          size: 'lg',
+          scope: $scope
+        });
 
-    $http.post('/api/visualizations/save', {type: type, visualizations: $scope.visualizations})
-    .success(function(data){
-      if(!data.error){
-        $modalInstance.close('saved');
-      }
-    });
-  }
-})
+        $scope.modalInstance.opened.then(function(){
+
+          $scope.save = function(user){
+            $scope.cuErrorMessages = [];
+            $scope.cuError = false;
```

Comment:
Please see my comment on line 686 for a proper solution for using this extra `bool`. This counts for all checks.